web/README.md

```diff
@@ -300,6 +300,18 @@ the root of this repository.
 Keep in mind that there should only be a single `yarn.lock` in this repository, here at the top level. If you add packages
 via `yarn workspace <workspace-name> add <package-name>`, it will create a `packages/<package-name>/yarn.lock` file, which should not be checked in.
 
+Dependencies should generally be added to the specific workspaces that use them. 
+For instance, if you need a new dependency for Web UI, add it to `packages/teleport/package.json`. 
+Similarly, if you're adding a new ESLint plugin, include it in `packages/build/package.json`.
+
+However, there are cases where dependencies should be added to the root `package.json`:
+
+1. Project-wide packages, such as `react`.
+2. Web UI Enterprise dependencies (these shouldn't be added to `e/web/teleport/package.json` 
+to avoid generating a different lockfile when e isn't cloned).
+3. Globally run CLI tools, like `prettier`.
+4. Dependencies imported in multiple workspaces, like `react-day-picker`.
```

Comment:
I feel like points 1 and 4 are kinda the same, no? React is considered "project-wide" only because Yarn doesn't give you the option to enforce a single package version per all workspaces, which I understand that pnpm does. I believe with pnpm it no longer needs to be considered a "project-wide" package, it's just a dep used by multiple workspaces.

Continuing this train of thought, aren't "dependencies imported in multiple workspaces" pretty much just shortcuts? If both teleport and teleterm need React, why couldn't they both define it in its deps? I believe this is exactly what we'd have to do if they were real packages published to a registry.

The only exception would be e-only deps, for which we use the shortcut of adding it to root package.json because any other way would have been more cumbersome.

I'm not necessarily saying that we should take action on this now (as in, add React to every workspace that uses it etc.). But maybe we could drop the concept of project-wide packages from this list, as I believe it's not a concept that's useful anymore.

Reply:
> Continuing this train of thought, aren't "dependencies imported in multiple workspaces" pretty much just shortcuts? If both teleport and teleterm need React, why couldn't they both define it in its deps? I believe this is exactly what we'd have to do if they were real packages published to a registry.

>The only exception would be e-only deps, for which we use the shortcut of adding it to root package.json because any other way would have been more cumbersome.

I looked again into the deps that we have in the root `package.json` and actually almost all of them are imported in `e`...
So perhaps we should say that dependencies imported in `e` should be kept in the root package.json. If a dependency is not imported in `e`, it should be kept in the workspace.
If a package is needed in multiple workspaces, it should be added in all of them.
For example, `events` should be declared in `teleport`, `teleterm` and `design` packages, instead of in the root.
In other words, I'm saying that we should drop points 1. and 4.

On a side note, the latest version of pnpm introduced [`catalogs` feature](https://pnpm.io/catalogs) that allow you to specify the package version globally. It would help with passing the same version to every workspace. OTOH, it still wouldn't solve the problem with e-only deps.

Reply:
So I'm thinking about something like this:

> Dependencies should generally be added to the specific workspaces that use them. 
> For instance, if you need a new dependency for Web UI, add it to `packages/teleport/package.json`. 
> Similarly, if you're adding a new ESLint plugin, include it in `packages/build/package.json`.
> If a dependency is imported in, for example, `packages/shared` and `packages/teleterm` 
> add it to both of them (that's what we'd have to do if they were real packages 
> published to a registry).
> 
> However, there are cases where dependencies should be added not in a workspace, but 
> in the root `package.json`:
> 1. Dependencies imported in `e` code (we can't declare dependencies in `e/web/teleport/package.json`
>    to avoid generating a different lockfile when `e` isn't cloned). 
>    For example, `react` - it is imported in every package (in `e` too), so it 
>    needs to be kept in the root.
> 2. CLI tools which are run from the root of the repo, like `prettier`.

Reply:
That sounds good for now. Catalogs seem like a good solution, but I'd worry if tools like dependabot or GitHub Alerts have support for it.

Reply:
>but I'd worry if tools like dependabot or GitHub Alerts have support for it.

Yeah, probably not in the close future.